File: TwoPointer/169.majority-element.ts
// @leet start
function majorityElement(nums: number[]): number {
  let major = nums[0];
  let count = 0;
  for (const v of nums) {
    if (v === major) {
      count++;
    } else {
      count--;
    }
    if (count < 0) {
      count = 0;
      major = v;
    }
  }
  return major;
}
// @leet end

File: TwoPointer/169.majority-element.py
from typing import List


# @leet start
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        major = nums[0]
        count = 0
        for i in nums:
            if i == major:
                count += 1
            else:
                count -= 1
            if count < 0:
                count = 0
                major = i
        return major


# @leet end

